### ISSUE TITLE
clang-tidy 14.0.6 gives a false-positive bugprone-use-after-move report in chained expressions

### ISSUE THREAD

**Issue by georgthegreat** [llvm/llvm-project#57758]
Consider the following code
```cpp
#include <cassert>
#include <vector>

struct test {
 test& foo(std::vector<int> const& v) { assert(v.size() == 1); return *this; }
 int bar(std::vector<int> v ) { return v.size(); }
};

int main() {
    std::vector<int> v = { 42 };
    return test{}.foo(v).bar(std::move(v));

}
```

clang-tidy 14.0.6 reports the following error:

```console
tidy14-bug/main.cpp:11:23: error: 'v' used after it was moved [bugprone-use-after-move,-warnings-as-errors]
    return test{}.foo(v).bar(std::move(v));
                      ^
tidy14-bug/main.cpp:11:30: note: move occurred here
    return test{}.foo(v).bar(std::move(v));
                             ^
tidy14-bug/main.cpp:11:23: note: the use and move are unsequenced, i.e. there is no guarantee about the order in which they are evaluated
    return test{}.foo(v).bar(std::move(v));
                      ^
```

The standard, however, clearly [defines the behavior](http://eel.is/c++draft/expr.call#8) for the case.
This false-positive breaks a lot of code in our codebase.

**Comment by georgthegreat** [comment]
It turns out that the exaple above is broken even with clang-tidy12.

**Comment by EugeneZelenko** [comment]
Could you please try 15 or  `main` branch? https://godbolt.org should be helpful.

**Comment by georgthegreat** [comment]
I can not import entire project into our infrastructure.
Could you, please, point to a commit or a set of commits fixing the problem?

**Comment by georgthegreat** [comment]
I can not find an option to run clang-tidy via godbolt.
It only supports compilation, and my problem is not related with compilation.

**Comment by EugeneZelenko** [comment]
`Add tool` button allow to do this.

**Comment by georgthegreat** [comment]
clang15 still have this problem:
https://godbolt.org/z/31Prdv1dK

so does clang12

**Comment by georgthegreat** [comment]
Corresponding StackOverflow discussion is [here](https://stackoverflow.com/questions/73729536/is-clang-tidy-version-14-right-with-bugprone-use-after-move-error-on-this-co).

**Comment by JonasToth** [comment]
the check predates `c++17` so very likely the guarantee from `c++-17` is not considered in the implementation.

**Comment by georgthegreat** [comment]
clang-tidy is capable of getting the standard version from cmdline.
This check, however, does not take the standard into account (I am able to reproduce the problem with `-std=c++20`).
